tests/command/test_navigation.rs

```diff
@@ -636,3 +636,40 @@ fn test_navigation_failed_to_check_out_commit() -> eyre::Result<()> {
 
     Ok(())
 }
+
+#[test]
+#[cfg(unix)]
+fn test_checkout_pty_branch() -> eyre::Result<()> {
+    let git = make_git()?;
+
+    git.init_repo()?;
+    git.commit_file("test1", 1)?;
+    git.detach_head()?;
+    git.commit_file("test2", 2)?;
+    git.run(&["checkout", "--detach", "master"])?;
+    git.commit_file("test3", 3)?;
+
+    run_in_pty(
+        &git,
+        &["branchless", "checkout"],
+        &[
+            PtyAction::WaitUntilContains("> "),
+            PtyAction::Write("master"),
+            PtyAction::WaitUntilContains("> master"),
```

Comment:
(If the tests are still failing) You may need to change this to `> 62fc20d2`. This is intended to match what `skim` thinks is the selected item -- e.g. what it will return when the carriage return is entered.

<img width="724" alt="Screen Shot 2021-11-05 at 6 37 38 AM" src="https://user-images.githubusercontent.com/706257/140519120-5d0cac6b-c1fd-499d-ad41-b50f562a7e60.png">

In this case, `> master` just verifies that the query was entered successfully, but there is a race condition in skim actually updating the displayed results. (You'd think key presses and the displayed results would be in ~lockstep, but in my experience there's a slight delay)

Reply:
Thanks, that fixed it! That probably saved me a lot of time :)